src/client/styles/style.css

```diff
@@ -0,0 +1,312 @@
+a {
+  text-decoration: none;
+}
+
+a#basket {
+  align-self: center;
+  justify-self: end;
+  position: relative;
+  top: 7px;
+}
+
+a#basket:after {
+  background-color: #f44336;
+  border-radius: 50%;
+  color: white;
+  content: attr(data-badge);
+  display: inline-block;
+  font-family: sans-serif;
+  font-size: 10px;
+  font-weight: bold;
+  left: -11px;
+  padding: 3px 3px 3px 3px;
+  position: relative;
+  text-align: center;
+  top: -26px;
+  width: 13px;
+}
+
+footer a {
+  color: #ccc;
+}
+
+footer a:hover {
+  color: white;
+}
+
+nav a {
+  color: white;
+  display: inline-block;
+  font-family: 'Roboto', sans-serif;
+  font-weight: 300;
+  padding: 10px 10px;
+}
+
+nav a:hover, nav a.current {
+  background-color: #eee;
+  color: #444;
+}
+
+body {
+  margin: 0;
+  padding: 0;
+}
+
+button {
+  background-color: #457FE2;
+  border: none;
+  border-radius: 2px;
+  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
+  color: #eee;
+  cursor: pointer;
+  font-family: 'Roboto', sans-serif;
+  font-size: 12px;
+  padding: 10px 15px;
+  text-transform: uppercase;
+}
+
+button:active {
+  background-color: #1976d2;
+  color: white;
+}
+
+button:hover {
+  background-color: #1976d2;
+  color: white;
+}
+
+button:focus {
+  box-shadow: 1px 1px 5px 0 rgba(0,0,0,.5);
+  outline: none;
+}
+
+button[disabled] {
+  background-color: rgb(30, 30, 30);
+  background-color: rgba(255, 255, 255, 0.12);
+  color: rgb(76, 76, 76);
+  color: rgba(255, 255, 255, 0.3);
+}
+
+main div {
+  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
+  display: grid;
+  grid-template-rows: max-content;
+  /* grid-auto-rows: max-content;*/
+  padding: 0 0 20px 0;
+  text-align: center;
+}
+
+footer {
+  background-color: #444;
+  box-shadow: 0 -2px 5px 0 rgba(0, 0, 0, 0.16);
+  color: white;
+  font-family: 'Roboto', sans-serif;
+  padding: 10px 30px 10px 30px;
+}
+
+header {
+  align-items: center;
+  /* background-color: #1976D2; */
+  background-color: #333;
+  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16);
+  display: grid;
+  grid-template-columns: 40px 100px auto 70px;
+  padding: 10px 0 10px 10px;
+  position: sticky;
+  top: 0;
+}
+
+h1 {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 100;
+  margin: 0;
+  padding: 30px;
+}
+
+header h1 {
+  color: white;
+  font-size: 20px;
+  font-weight: 300;
+  margin: 0;
+  padding: 0 0 2px 0;
+  white-space: nowrap;
+}
+
+header h1::before {
+  content: '';
+}
+
+h2 {
+  font-family: 'Roboto', sans-serif;
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 1.3em;
+  /*  min-height: 3em;*/
+  margin: 0 0 13px 0;
+  padding: 0 10px;
+  white-space: nowrap;
+}
+
+main img {
+  margin: 0 0 20px 0;
+  width: 100%;
+}
+
+input#search {
```

Comment:
I think IDs and ID selectors make sense: 'there's only one of these', as opposed to `input.search`, which to me implies that there's more than one. 

I know the CSSLint arguments against this, but I don't agree :). In practice, I think styles for an ID selector are very rarely 'reused later' — or you create a class.

Do you have other reasons? I know that many web developer now just assume ID Selectors Are Bad, so we might be better off avoiding grief...

Reply:
Main reason I tend to avoid ids in css is that I reserve them for js usage, rather than worry about their uniqueness. This way js dev knows they can change/come up with ids if needed without breaking css rules. 

I'm not going to start a war about this though;) So it's up to you.

If we keep using ids in css, input#search and #search are the same. We might want to remove input for briefness, unless we want to be less-maintainable but more-readable.